set.mm

```diff
@@ -481233,6 +481329,394 @@ such that every prime ideal contains a prime element (this is a
+      QUKQSTUFBUIJUFBAUAGUICAUBUCUDUE $.
+
+    ${
+      $d I i j $.  $d P j $.  $d R i j $.  $d V i $.
+      rspectset.1 $e |- I = ( LIdeal ` R ) $.
+      rspectset.2 $e |- J = ran ( i e. I |-> { j e. I | -. i C_ j } ) $.
+      $( Topology component of the spectrum of a ring.  (Contributed by Thierry
+         Arnoux, 2-Jun-2024.) $)
+      rspectset $p |- ( R e. Ring -> J = ( TopSet ` S ) ) $=
+        ( crg wcel cidlsrg cfv cts cprmidl cress co cvv wceq eqid fvex resstset
+        ax-mp idlsrgtset crspec rspecval syl5eq fveq2d 3eqtr4a ) AJKZALMZNMZUKA
+        OMZPQZNMZFBNMUMRKULUOSAOUAUMUKUNULRUNTULTUBUCAUKCDEFJUKTHIUDUJBUNNUJBAU
+        EMUNGAUFUGUHUI $.
+    $}
+
+    ${
+      $d I i j k $.  $d I i j x y $.  $d J x $.  $d P i j k $.  $d P x y $.
+      $d R i j k $.  $d R x $.
+      rspectopn.1 $e |- I = ( LIdeal ` R ) $.
+      rspectopn.2 $e |- P = ( PrmIdeal ` R ) $.
+      rspectopn.3 $e |- J = ran ( i e. I |-> { j e. P | -. i C_ j } ) $.
+      $( The topology component of the spectrum of a ring.  (Contributed by
+         Thierry Arnoux, 4-Jun-2024.) $)
+      rspectopn $p |- ( R e. Ring -> J = ( TopOpen ` S ) ) $=
+        ( vy wcel ctopn cfv co cv eqid wceq cvv crg cidlsrg crest wss crab cmpt
+        vx wn crn cress crspec cprmidl rspecval oveq2i 3eqtr4g resstopn syl6eqr
+        fveq2d cts idlsrgtset cbs cpw wral wa clidl fvexi rabex simp2 idlsrgbas
+        a1i w3a adantr 3ad2ant1 eleqtrd rabssdv elpwd ralrimiva eqsstrrd topnid
+        rnmptss syl eqtrd oveq1d wrex cin mptex elrest mp2an rgenw ineq1 eqeq2d
+        wb rnex rexrnmptw ax-mp inrab2 prmidlssidl 3sstr4g sseqin2 sylib syl5eq
+        rabeqdv rexbidv syl5bb eleq2i elrnmpti bitri syl6bbr eqrdv 3eqtr2rd ) B
+        UAMZCNOZBUBOZNOZAUCPZDFDQZEQZUDUHZEFUEZUFZUIZAUCPZGXKXLXMAUJPZNOXOXKCYC
+        NXKBUKOXMBULOZUJPCYCBUMHAYDXMUJJUNUOURAYCXNXMYCRXNRUPUQXKYAXNAUCXKYAXMU
+        SOZXNBXMDEFYAUAXMRZIYARUTZXKYEXMVAOZVBZUDYEXNSXKYEYAYIYGXKXSYIMZDFVCYAY
+        IUDXKYJDFXKXPFMZVDZXSYHTXSTMZYLXREFFBVEIVFZVGZVJYLXREFYHYLXQFMZXRVKXQFY
+        HYLYPXRVHYLYPFYHSZXRXKYQYKBXMFUAYFIVIVLVMVNVOVPVQDFXSYIXTXTRZVTWAVRYHYE
+        XMYHRYERVSWAWBWCXKUGYBGXKUGQZYBMZYSXREAUEZSZDFWDZYSGMZYTYSLQZAWEZSZLYAW
+        DZXKUUCYATMATMYTUUHWLXTDFXSYNWFWMABULJVFZLYSAYATTWGWHUUHYSXSAWEZSZDFWDZ
+        XKUUCYMDFVCUUHUULWLYMDFYOWIUUGUUKDLFXSXTTYRUUEXSSUUFUUJYSUUEXSAWJWKWNWO
+        XKUUKUUBDFXKUUJUUAYSXKUUJXREFAWEZUEUUAXREFAWPXKXREUUMAXKAFUDUUMASXKYDBV
+        EOAFBWQJIWRAFWSWTXBXAWKXCXDXDUUDYSDFUUAUFZUIZMUUCGUUOYSKXEDFUUAYSUUNUUN
+        RXREAUUIVGXFXGXHXIXJ $.
+    $}
+  $}
+
+  ${
+    zarclsx.1 $e |- V = ( i e. ( LIdeal ` R )
+                                    |-> { j e. ( PrmIdeal ` R ) | i C_ j } ) $.
+
+    ${
+      $d .0. i j $.  $d P i $.  $d R i j $.
+      zarcls0.1 $e |- P = ( PrmIdeal ` R ) $.
+      zarcls0.2 $e |- .0. = ( 0g ` R ) $.
+      $( The closure of the identity ideal in the Zariski topology.
+         (Contributed by Thierry Arnoux, 16-Jun-2024.) $)
```

Comment:
What "is" a Zariski topology? Is it ` V `? According to Wikipedia, "the Zariski topology is a topology defined on geometric objects called varieties.". A short explanation, and a reference to literature (at least to Wikipedia) would be very helpful.

Reply:
The topology itself is a collection of _open sets_. This is the `J` in ~ rspectopn and ~ zartop.

But the textbook definitions all start from _closed sets_. This is also the case for [Wikipedia](https://en.wikipedia.org/wiki/Zariski_topology#Spectrum_of_a_ring).

The closed sets are generated from the ideals using the function `V`, so that the closed sets are actually `ran V`: see ~ zatopn .

The bridge between the closed set `ran V` and the open sets ` J ` is made in ~ zarcls, simply taking the complements.

Reply:
I've added a short explanation in the chapter's heading.